xarray/tests/test_units.py

```diff
@@ -78,16 +175,18 @@ def array_extract_units(obj):
     if isinstance(obj, (xr.Variable, xr.DataArray, xr.Dataset)):
         obj = obj.data
 
-    try:
-        return obj.units
-    except AttributeError:
+    lib = get_unit_lib(obj)
+    if lib is not None:
+        return lib.get_unit(obj)
+    else:
         return None
```

Comment:
I'm not sure how to implement a `get_unit()` method on a "no unit library" class, and I don't think it would make sense to sub-class `UnitInfo` for something that doesn't actually have units.

Reply:
I was suggesting to use a dummy class that defines its `get_unit` method to always return `None`:
```python
class not_a_quantity:
    def get_unit(self, obj):
        return None

unit_libs = {...}  # does not contain `no_quantity`

a = 1
b = ureg.Quantity(1, "m")

def get_unit(obj):
    lib = units_libs.get(type(obj), not_a_quantity)
    return lib.get_unit(obj)

get_unit(b)  # Unit("m")
get_unit(a)  # None
```

In other words, by making use of the `dict.get` default we can avoid the additional comparison with `None`, where `None` is the default value for `dict.get`. Not sure how much of a difference that makes, though.